packages/core/src/common/metrics.ts

```diff
@@ -255,72 +260,87 @@ export async function getHistoricalSyncProgress(metrics: MetricsService) {
-    ([network, totalBlocks]) => {
-      const cachedBlocks = cachedBlocksMetric[network];
-      const completedBlocks = completedBlocksMetric[network] ?? 0;
-
-      // If cachedBlocks is not set, setup is not complete.
-      if (cachedBlocks === undefined) {
-        return {
-          networkName: network,
-          totalBlocks,
-          completedBlocks,
-        };
+  const extractMetric = (
+    metric: prometheus.MetricObjectWithValues<
+      prometheus.MetricValue<"network">
+    >,
+    network: string,
+  ) => {
+    return metric.values.find((m) => m.labels.network === network)?.value;
+  };
+
+  const totalBlocksMetric = await metrics.ponder_historical_total_blocks.get();
+  const cachedBlocksMetric =
+    await metrics.ponder_historical_cached_blocks.get();
+  const completedBlocksMetric =
+    await metrics.ponder_historical_completed_blocks.get();
+  const syncBlockMetric = await metrics.ponder_sync_block.get();
+  const syncIsRealtimeMetrics = await metrics.ponder_sync_is_realtime.get();
+  const syncIsCompleteMetrics = await metrics.ponder_sync_is_complete.get();
+
+  const requestCount: { [network: string]: number } = {};
+  const rpcRequestMetrics = await metrics.ponder_rpc_request_duration.get();
+  for (const m of rpcRequestMetrics.values) {
+    if (m.metricName === "ponder_rpc_request_duration_count") {
+      if (requestCount[m.labels.network!] === undefined) {
+        requestCount[m.labels.network!] = 0;
       }
+      requestCount[m.labels.network!] += m.value;
+    }
+  }
 
-      const progress =
-        totalBlocks === 0 ? 1 : (completedBlocks + cachedBlocks) / totalBlocks;
-
-      const elapsed = syncDurationSum[network]!;
-      const total = elapsed / (completedBlocks / (totalBlocks - cachedBlocks));
-      // The ETA is low quality if we've completed only one or two blocks.
-      const eta = completedBlocks >= 3 ? total - elapsed : undefined;
-
-      return {
-        networkName: network,
-        totalBlocks,
-        cachedBlocks,
-        completedBlocks,
-        progress,
-        eta,
-      };
-    },
-  );
+  for (const [networkName, count] of Object.entries(requestCount)) {
+    if (rps[networkName] === undefined) {
+      rps[networkName] = [{ count, timestamp: Date.now() }];
+    } else {
+      rps[networkName]!.push({ count, timestamp: Date.now() });
+    }
 
-  const totalBlocks = networks.reduce((a, c) => a + c.totalBlocks, 0);
-  const cachedBlocks = networks.reduce((a, c) => a + (c.cachedBlocks ?? 0), 0);
-  const completedBlocks = networks.reduce(
-    (a, c) => a + (c.completedBlocks ?? 0),
-    0,
-  );
-  const progress =
-    totalBlocks === 0 ? 1 : (completedBlocks + cachedBlocks) / totalBlocks;
+    if (rps[networkName]!.length > 100) {
+      rps[networkName]!.shift();
+    }
+  }
 
-  return {
-    overall: progress,
-    networks,
-  };
+  return totalBlocksMetric.values.map(({ value, labels }) => {
+    const totalBlocks = value;
```

Comment:
Could make this more compact by doing `const network = labels.network as string` here at the top